keep original extension for data download

All4Gis opened the pull request:
Currently download files are encapsulated in a tar. We don't want that,it's strange for the user

We want to keep the original format, that is, the downloaded file has to keep its structure, extension, if we download it from here or if we download it from curl for example. This PR fix this.


**Changes**

- The downloaded data keep the same extension as the original download via curl for example or https.
- A new test has been added to test the download of a single file. `test_download_data_single_file`

keflavich commented on astroquery/esa/xmm_newton/core.py:
```diff
@@ -101,13 +103,21 @@ def download_data(self, observation_id, filename=None, verbose=False,
             log.info(link)
 
-        response = self._request('GET', link, save=True, cache=True)
+        response = self._request('GET', link, save=False, cache=True)
```
can this be achieved with a `'HEAD'` request?

All4Gis replied:
What exactly does that do?

keflavich replied:
HTTP HEAD requests: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD are a way to just get the headers, not the data - it is presumably quite a bit cheaper than a full `GET` request.

All4Gis replied:
Ah ok, If we only get the header to get the headers we would make a` 'HEAD'`, but we want to download it not only to get his name.We can implement a GET and a HEAD for get the filename, but the GET method already returns everything we need

keflavich replied:
ah, of course, you're right - you're doing everything in one request.  The approach you took here is totally fine.

All4Gis replied:
But I think you're right, I'm gonna change a few things

All4Gis commented:
Any updates on this? Travis errors etc, are not related to PR

ceb8 commented:
Can you do a rebase to make the tests pass and get rid of the astropy_helpers changes.

Also, please add a changelog entry.

All4Gis commented:
> Can you do a rebase to make the tests pass and get rid of the astropy_helpers changes.
Removed the helper's changes
> 
> Also, please add a changelog entry.
Added changelog

Waiting for the rebase

bsipocz commented:
@All4Gis - what do you exactly mean by "Waiting for rebase"? are you in the process of doing it or would like to get help with it?

All4Gis commented:
> @All4Gis - what do you exactly mean by "Waiting for rebase"? are you in the process of doing it or would like to get help with it?
In the process of rebase our fork. Before rebase I have to see the state of my colleagues' forks. Don't worry, I will soon update this PR

All4Gis commented:
> @All4Gis - what do you exactly mean by "Waiting for rebase"? are you in the process of doing it or would like to get help with it?

It's already updated and passing travis

bsipocz commented:
@All4Gis - this hasn't been rebased, there is still the one commit that relevant to this PR but the other 9 are not (e.g. pulling in merge commits are not our development workflow).

I go ahead and rebase it myself and add a changelog entry prior merging. 

My one suggestion would be to work from your own forks instead of trying to keep a parallel organizational fork updated. If the org fork is necessary for internal ESA tracking, I think I could come up with something that would cause less troubles with the current setup (basically that fork could track your branches, and not the other way around). 

In the meantime the advanced exercises here cover most of the complications of working with multiple forks here: https://learngitbranching.js.org/

bsipocz commented:
The rebase here was:

```
git rebase -i astropy/master
# keep only the second commit, with message "keep the original extension/ add new test", exit the editor
git push esdc-esac-esa-int HEAD:feat-XXX-keep-ori-extension -f
```

I made a copy of the pre-rebase branch, in case you want to compare the two:
https://github.com/bsipocz/astroquery/tree/copy_preprebase_feat-XXX-keep-ori-extension

bsipocz commented:
Thanks @All4Gis!
